libtransmission/global-ip-cache.cc

```diff
@@ -148,7 +148,7 @@ tr_global_ip_cache::tr_global_ip_cache(Mediator& mediator_in)
 
 std::unique_ptr<tr_global_ip_cache> tr_global_ip_cache::create(tr_global_ip_cache::Mediator& mediator_in)
 {
-    return std::unique_ptr<tr_global_ip_cache>(new tr_global_ip_cache(mediator_in));
+    return std::unique_ptr<tr_global_ip_cache>{ new tr_global_ip_cache(mediator_in) };
```

Comment:
Why not `std::make_unique<>()`?

Reply:
Because the constructor is private.

Reply:
I think the usual way to handle this is to make the ctor public, but with an additional parameter of private type.

OTOH I don't see any secret in the way this object is contructed, the `create()` method parameters match ctor parameters exactly, so why make the ctor private and need this method in the first place?..

Reply:
I think I was just trying to keep the API consistent with the other existing classes, because a lot of them had `create()` too, but you're right that this class doesn't need it.

I'll make the refactor later.

Reply:
760c812c805040c51161cbe71881838889f245fc